tfhe/src/test_user_docs.rs

```diff
@@ -1,102 +1,116 @@
-);
-// fine_grained_api/Boolean
-doctest!(
-    "../docs/fine_grained_api/Boolean/operations.md",
-    booleans_operations
-);
-doctest!(
-    "../docs/fine_grained_api/Boolean/parameters.md",
-    booleans_parameters
-);
-doctest!(
-    "../docs/fine_grained_api/Boolean/serialization.md",
-    booleans_serialization
-);
-doctest!(
-    "../docs/fine_grained_api/Boolean/readme.md",
-    booleans_tutorial
-);
+    // fine_grained_api/shortint
+    doctest!(
+        "../docs/fine_grained_api/shortint/operations.md",
+        shortint_operations
+    );
 
-// fine_grained_api/shortint
-doctest!(
-    "../docs/fine_grained_api/shortint/operations.md",
-    shortint_operations
-);
+    doctest!(
+        "../docs/fine_grained_api/shortint/parameters.md",
+        shortint_parameters
+    );
+    doctest!(
+        "../docs/fine_grained_api/shortint/serialization.md",
+        shortint_serialization
+    );
+    doctest!(
+        "../docs/fine_grained_api/shortint/readme.md",
+        shortint_tutorial
+    );
 
-doctest!(
-    "../docs/fine_grained_api/shortint/parameters.md",
-    shortint_parameters
-);
-doctest!(
-    "../docs/fine_grained_api/shortint/serialization.md",
-    shortint_serialization
-);
-doctest!(
-    "../docs/fine_grained_api/shortint/readme.md",
-    shortint_tutorial
-);
+    // fine_grained_api/integer
+    doctest!(
+        "../docs/fine_grained_api/integer/operations.md",
+        integer_operations
+    );
+    doctest!(
+        "../docs/fine_grained_api/integer/serialization.md",
+        integer_serialization_tuto
+    );
+    doctest!(
+        "../docs/fine_grained_api/integer/readme.md",
+        integer_first_circuit
+    );
 
-// fine_grained_api/integer
-doctest!(
-    "../docs/fine_grained_api/integer/operations.md",
-    integer_operations
-);
-doctest!(
-    "../docs/fine_grained_api/integer/serialization.md",
-    integer_serialization_tuto
-);
-doctest!(
-    "../docs/fine_grained_api/integer/readme.md",
-    integer_first_circuit
-);
+    // core_crypto
+    doctest!(
+        "../docs/core_crypto/presentation.md",
+        core_crypto_presentation
+    );
+    doctest!("../docs/core_crypto/tutorial.md", core_crypto_tutorial);
 
-// core_crypto
-doctest!(
-    "../docs/core_crypto/presentation.md",
-    core_crypto_presentation
-);
-doctest!("../docs/core_crypto/tutorial.md", core_crypto_tutorial);
+    // Tutorials
+    doctest!(
+        "../docs/tutorials/ascii_fhe_string.md",
+        ascii_fhe_string_tutorial
+    );
+    doctest!("../docs/tutorials/parity_bit.md", parity_bit_tutorial);
+}
 
-// Tutorials
-doctest!(
-    "../docs/tutorials/ascii_fhe_string.md",
-    ascii_fhe_string_tutorial
-);
-doctest!("../docs/tutorials/parity_bit.md", parity_bit_tutorial);
+#[cfg(feature = "gpu")]
+fn test_gpu_doc() {
+    doctest!("../docs/how_to/run_on_gpu.md", how_to_run_on_gpu);
+}
+
+fn main() {
+    #[cfg(not(feature = "gpu"))]
+    test_cpu_doc();
+    #[cfg(feature = "gpu")]
+    test_gpu_doc()
+}
```

Comment:
main ?

Reply:
I think we are not running any doctests here ? as main is a private function called nowhere ?